test/integration/other/migrations.js

```diff
@@ -1054,4 +1054,106 @@ testMigration('20240914-02-remove-orphaned-client-audits.js', () => {
+        .set('Content-Type', 'text/xml')
+        .expect(200);
+
+      // Upload another form that needs updating
+      await asAlice.post('/v1/projects/1/forms')
+        .send(testData.forms.updateEntity.replace('id="updateEntity"', 'id="updateEntity2"'))
+        .expect(200);
+
+      // Upload an entity form that doesn't really need updating but does have 'update' in the actions column
+      await asAlice.post('/v1/projects/1/forms')
+        .send(testData.forms.offlineEntity)
+        .expect(200);
+
+      // Upload an entity form that does not need updating
+      await asAlice.post('/v1/projects/1/forms')
+        .send(testData.forms.simpleEntity)
+        .expect(200);
+
+      // Deleted forms and projects
+      // Upload another form that needs updating but will be deleted
+      await asAlice.post('/v1/projects/1/forms')
+        .send(testData.forms.updateEntity.replace('id="updateEntity"', 'id="updateEntityDeleted"'))
+        .expect(200);
+
+      await asAlice.delete('/v1/projects/1/forms/updateEntityDeleted')
+        .expect(200);
+
+      // Create a new project
+      const newProjectId = await asAlice.post('/v1/projects')
+        .send({ name: 'NewDeletedProject' })
+        .expect(200)
+        .then(({ body }) => body.id);
+
+      // Upload a form that needs updating to the new project
+      await asAlice.post(`/v1/projects/${newProjectId}/forms?publish=true`)
+        .send(testData.forms.updateEntity)
+        .expect(200);
+
+      // Delete the new project
+      await asAlice.delete(`/v1/projects/${newProjectId}`);
+
+      // Mark forms for upgrade
+      await up();
+
+      // Check: 2 forms that need upgradin and 1 2024.1 form that it wont harm to run through the worker
+      const audits = await container.oneFirst(sql`select count(*) from audits where action = 'upgrade.process.form.entities_version'`);
+      audits.should.equal(3);
+
+      // Run upgrade
+      await exhaust(container);
+
+      await asAlice.get('/v1/projects/1/forms/updateEntity.xml')
+        .then(({ text }) => {
+          text.should.equal(`<?xml version="1.0"?>
+<h:html xmlns="http://www.w3.org/2002/xforms" xmlns:h="http://www.w3.org/1999/xhtml" xmlns:jr="http://openrosa.org/javarosa" xmlns:entities="http://www.opendatakit.org/xforms">
+  <h:head>
+    <model entities:entities-version="2024.1.0">
+      <instance>
+        <data id="updateEntity" orx:version="2.0[upgrade]">
+          <name/>
+          <age/>
+          <hometown/>
+          <meta>
+            <entity dataset="people" id="" update="" baseVersion="" trunkVersion="" branchId="">
+              <label/>
+            </entity>
+          </meta>
+        </data>
+      </instance>
+      <bind nodeset="/data/name" type="string" entities:saveto="first_name"/>
+      <bind nodeset="/data/age" type="int" entities:saveto="age"/>
+    </model>
+  </h:head>
+</h:html>`);
+        });
+
```

Comment:
Also test that the draft of updateEntity is version 3.0[upgrade]. 
Also test that the other updateEntity2 form is upgraded to show the after state of the 3 forms that were processed by the worker. 

Check that the right events are being inserted as well.